src/pipert2/core/base/routine.py

```diff
@@ -188,3 +189,11 @@ def notify_event(self, event_name: str, **event_params) -> None:
         """
 
         self.event_notifier(event_name, **event_params)
+
+    def join(self):
+        """Block until all routine thread terminates
+
+        """
+
+        if self.stop_event.is_set():
+            self.runner.join()
```

Comment:
what does this runner do?

Reply:
For now, it starts the routine's thread.
In the future it might be a task runner (asyncio) or a process runner (multiprocessing)